server/src/main/java/org/elasticsearch/cluster/routing/package-info.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/**
+ * <p>Cluster routing package provides routing information and manages shard
+ * allocations in cluster. The routing part contains two different views on
+ * shards in cluster {@link org.elasticsearch.cluster.routing.RoutingTable} and
+ * {@link org.elasticsearch.cluster.routing.RoutingNodes}. RoutingTable provides
+ * view from index to shard to node, and RoutingNodes provides view from node to
+ * shard. Shard allocation is a process of assigning and moving shards between
+ * nodes. For more details about allocation see {@link
+ * org.elasticsearch.cluster.routing.allocation}.</p>
+ *
+ * <b>Routing Tables</b>
+ *
+ * <p>Routing tables provide a view on global cluster state from index
+ * perspective. It's a mapping from indices to shards and shards to nodes, where
+ * relationship between shard and node might not exists if shard allocation is
+ * not possible. Routing table represents desired shards layout and current
+ * state of shards.</p>
```

Comment:
I don't understand this last sentence. The previous sentence states that a shard-to-node mapping may not exist (if the shard has not been allocated yet). And then we're talking about desired shard allocation being in the routing table?

Reply:
"desired" word might be confusing, its desired from allocator perspective. If allocator cannot find right place for shard it will be unassigned.

Reply:
So where is the desired allocation information saved? Presumably not the same place as the actual mappings? Or is it all in one data structure and there is some kind of flag saying that the shard in question is not allocated?

Reply:
I would suggest to rephrase it to `Routing table represents current state of shards`

Reply:
@DiannaHohensee, my understanding that allocator will create assign shards to nodes and produce new routing table. When node receive cluster update it will see which shards are allocated there and will start initialization process - shard state machine. So routing table is a desired layout in terms of shards assignment and current state of a shard at the same time.
And I agree with @idegtiarenko, probably better to keep it simple here. And move allocation part into "future" cluster.routing.allocation package-info